### PR TITLE
[MOB-5832] update delegate

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 Jira Ticket(s)

* [MOB-5832](https://iterable.atlassian.net/browse/MOB-5832)

## ✏️ Description

* adds the ability to add UI objects as a delegate for updated messages

[MOB-5832]: https://iterable.atlassian.net/browse/MOB-5832?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swift-sdk/Internal/EmbeddedMessagingManager.swift [hunk 1]
```diff
@@ -148,5 +148,5 @@ class EmbeddedMessagingManager: NSObject, IterableEmbeddedMessagingManagerProtoc
     
     private var messages: [IterableEmbeddedMessage] = []
     
-    private var listeners: [IterableEmbeddedMessagingUpdateDelegate] = []
+    private var listeners: NSHashTable<IterableEmbeddedMessagingUpdateDelegate> = NSHashTable(options: [.weakMemory])
 }
```
Comment (evangreer91) [inline swift-sdk/Internal/EmbeddedMessagingManager.swift:152]: Is there a reason for using weak memory?

Comment (roninopf) [reply swift-sdk/Internal/EmbeddedMessagingManager.swift:152]: So if the UI object is removed from memory, we don't inadvertently keep it around (memory leak). We don't have a need to have a direct reference to anything that goes in there.

**Review by evangreer91 — COMMENTED** [review]
Some small comments/clarification

File: tests/embedded-messaging-tests/EmbeddedMessagingProcessorTests.swift [hunk 2]
```diff
@@ -7,7 +7,18 @@ import XCTest
...
+        
+        let processor = EmbeddedMessagingProcessor(currentMessages: currentMessages,
+                                                   fetchedMessages: fetchedMessages)
+        
+        XCTAssertEqual(processor.processedMessagesList().map { $0.metadata.id },
+                       ["a", "b", "c", "d", "e"])
+    }
+    
+    func testMessageIdsToTrackDelivery() {
         let currentMessages = makeBlankMessagesList(with: ["a", "b", "c"])
```
[the remark this answers is not in the document]
Comment (roninopf) [reply tests/embedded-messaging-tests/EmbeddedMessagingProcessorTests.swift:32]: Weird, I rewrote this test, I don't know why it's still showing up. It does now test for the fetched messages, yes.

[the remark this answers is not in the document]
Comment (roninopf) [reply tests/embedded-messaging-tests/EmbeddedMessagingProcessorTests.swift:32]: Looks like it's gone now!

**Review by evangreer91 — APPROVED** [review]
LGTM

**Pull request merged**